enterprise/server/cmd/goinit/main.go

```diff
@@ -276,6 +280,10 @@ func main() {
 		die(vmvfs.Run())
 		return
 	}
+	if *isVMDNS {
+		die(runVMDNSServer(rootContext))
+		return
+	}
```

Comment:
I think we might want to run the `vmdns` server as part of the `vmexec` server. That way, the lifecycle of the DNS server is tied to the `vmexec` server, so that if the DNS server crashes, that means that `vmexec` also crashed, which means that users will never see DNS errors (instead they'll get Unavailable errors due to the `vmexec` healthcheck failing). Another benefit is that the `vmexec` server has an `Init()` RPC that we could use to refresh the DNS information from MMDS, instead of having to invalidate all snapshots when we change the DNS configuration.

Reply:
Thanks for explaining! I understand better now. I think we've been overloading the term `vmexec`. It refers both to a server, which listens for GRPC requests on a port, and also a child process where that server runs. I didn't realize you were referring to the second. I refactored the code to run both the vmexec server and vmdns server in a single process, which is what I believe you were suggesting

Reply:
Also I think this solution addresses the race condition, but talking it out just to sanity check:
* There's no longer a race between two separate processes, because there's only one
* There could still be a case where the DNS server returns an error (and the entire process doesn't die). There could be a brief window of time where the vmexec server is still serving requests before the process is killed. Then /etc/resolv.conf would fallback to another nameserver after a timeout. However I think it's really unlikely that this timeout would happen before the process was killed by log.Fatal() when the DNS server died